### PR TITLE
Timer

### PR DESCRIPTION
**Pull request by erraticallyrational**
Finished timer and cleaned up code.

### CODE REVIEW

File: Werk/WorkoutCreation/ExerciseData/Interval/IntervalPicker.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+//  IntervalPicker.swift
+//  Werk
+//
+//  Created by Shaquil Campbell on 7/16/23.
+//
+
+import SwiftUI
+
+struct IntervalPicker: View {
+    //    @Binding var show: Bool
```
Comment (kinnectusAdmin) [inline Werk/WorkoutCreation/ExerciseData/Interval/IntervalPicker.swift:11]: Remove commented code

File: Werk/WorkoutCreation/Interval/IntervalPicker.swift [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+//  IntervalPicker.swift
+//  Werk
+//
+//  Created by Shaquil Campbell on 7/16/23.
+//
+
+import SwiftUI
+
+struct IntervalPicker: View {
+    //    @Binding var show: Bool
```
Comment (kinnectusAdmin) [inline Werk/WorkoutCreation/Interval/IntervalPicker.swift:11]: Remove commented code

File: Werk/WorkoutCreation/WorkoutCreationEditViewModel.swift [hunk 3]
```diff
@@ -25,15 +25,15 @@ class WorkoutCreationViewModel:Identifiable, ObservableObject {
...
+    init(workout: WorkoutBlueprint = WorkoutBlueprint.initial, service: DataStorageServiceIdentity = DataStorageService()) {
         self.service = service
         self.workout = workout
         self.workoutNameBinding = .init(get: provideWorkoutName, set: updateWorkoutName)
         $workout.map(\.warmup.duration).map({durationOfWorkout(duration: Double($0))}).assign(to: &$warmupDuration)
         $workout.map(\.cooldown.duration).map({durationOfWorkout(duration: Double($0))}).assign(to: &$cooldownDuration)
         $workout.map(\.intervals).assign(to: &$intervals)
-        $workout.map(\.highIntensity.duration).map({durationOfWorkout(duration: Double($0))}).assign(to: &$highIntensityDuration)
-        $workout.map(\.lowIntensity.duration).map({durationOfWorkout(duration: Double($0))}).assign(to: &$lowIntensityDuration)
+//        $workout.map(\.highIntensity.duration).map({durationOfWorkout(duration: Double($0))}).assign(to: &$highIntensityDuration)
```
Comment (kinnectusAdmin) [inline Werk/WorkoutCreation/WorkoutCreationEditViewModel.swift:35]: Remove.

File: Werk/Timer/TimerView.swift [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+//
+
+import SwiftUI
+
+struct TimerView: View {
+    @Environment(\.presentationMode) var presentationMode
+    @ObservedObject var viewModel:TimerViewModel = TimerViewModel(workout: WorkoutBlueprint.initial)
+    @State private var showingSheet = false
+    
+    //    let interval: Int = 0
```
[the remark this answers is not in the document]
Comment (erraticallyrational) [reply Werk/Timer/TimerView.swift:15]: removed

File: Werk/Timer/TimerView.swift [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+                            .padding()
+                            .foregroundColor(.white)
+                            .cornerRadius(10)
+                    }
+                    Spacer()
+                    //displays current set
+                    VStack{
+                        Text("\(viewModel.currentPhaseName)")
+                            .multilineTextAlignment(.center)
+                        Text("\(viewModel.currentPhaseIndex+1)/\(viewModel.workoutBlocks.count) Set")
```
[the remark this answers is not in the document]
Comment (erraticallyrational) [reply Werk/Timer/TimerView.swift:87]: changed to
viewModel.displayedPhase
viewModel.displayedSetInfo

File: Werk/Timer/TimerView.swift [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+                            .foregroundColor(.white)
+                            .cornerRadius(10)
+                    }
+                }
+                
+    
+                HStack{
+                    Spacer()
+                    //shows the total amont of time that has elapsed during the workout
+                    Text("\(viewModel.convertedElapsedTime) \nElapsed")
```
[the remark this answers is not in the document]
Comment (erraticallyrational) [reply Werk/Timer/TimerView.swift:112]: changed to
viewModel.displayedElapsedTime

File: Werk/Timer/TimerView.swift [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+                        .multilineTextAlignment(.center)
+                    
+                    Spacer()
+                    //shows the total duration of the workout
+                    Text("\(viewModel.convertedTotalDuration) \nRemaining")
```
[the remark this answers is not in the document]
Comment (erraticallyrational) [reply Werk/Timer/TimerView.swift:117]: changed to
viewModel.displayedTimeRemaining

File: Werk/Timer/TimerView.swift [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+                            .rotationEffect(.degrees(-90))
+
+                            
+                        //  start/resume button
+                        Button {
+                            viewModel.didPressStartorResume()
+                        } label: {
+                            ZStack {
+                                Circle()
+                                    .fill(Color.cyan)//Will remove after button code is set
```
[the remark this answers is not in the document]
Comment (erraticallyrational) [reply Werk/Timer/TimerView.swift:166]: removed

File: Werk/Timer/UpperTimerView.swift [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+                .ignoresSafeArea()
+            VStack {
+                HStack {
+                    Image(systemName: "xmark")
+                         .font(.system(size: 25))
+                         .foregroundColor(.white)
+                         .padding(.all, 5)
+                         .background(Color.black.opacity(0.0))
+                         .clipShape(Circle())
+                         .accessibility(label:Text("Close"))
```
[the remark this answers is not in the document]
Comment (erraticallyrational) [reply Werk/Timer/UpperTimerView.swift:26]: file no longer present

File: Werk/Timer/UpperTimerView.swift [hunk 10]
```diff
@@ -0,0 +1,101 @@
+//
```
[the remark this answers is not in the document]
Comment (erraticallyrational) [reply Werk/Timer/UpperTimerView.swift:1]: file no longer present our versions may be different

File: Werk/WorkoutCreation/ExerciseData/Interval/IntervalPicker.swift (hunk 1 above)
Comment (erraticallyrational) [reply Werk/WorkoutCreation/ExerciseData/Interval/IntervalPicker.swift:11]: removed

File: Werk/WorkoutCreation/Interval/IntervalPicker.swift (hunk 2 above)
Comment (erraticallyrational) [reply Werk/WorkoutCreation/Interval/IntervalPicker.swift:11]: removed

File: Werk/WorkoutCreation/WorkoutCreationEditViewModel.swift (hunk 3 above)
Comment (erraticallyrational) [reply Werk/WorkoutCreation/WorkoutCreationEditViewModel.swift:35]: removed

**Pull request merged**